enclave-runner/src/usercalls/mod.rs

```diff
@@ -681,21 +734,52 @@ impl RunningTcs {
         local_addr: Option<&mut OutputBuffer>,
         peer_addr: Option<&mut OutputBuffer>,
     ) -> IoResult<Fd> {
-        let addr = str::from_utf8(addr).map_err(|_| IoErrorKind::ConnectionRefused)?;
-        let stream = TcpStream::connect(addr)?;
-        if let Some(local_addr) = local_addr {
-            match stream.local_addr() {
-                Ok(local) => local_addr.set(local.to_string().into_bytes()),
-                Err(_) => local_addr.set(&b"error"[..]),
+        let mut l = String::new();
+        let mut p = String::new();
+        let o = 
+        {
+            let la = local_addr.as_ref().map(|_| &mut l);
+            let pa = peer_addr.as_ref().map(|_| &mut p);
+
+            self.enclave.usercall_ext.connect_stream(addr, la, pa)?
+        };
+
+        let stream = 
+        {
```

Comment:
nit: no need for this block, just start `let stream = match o {`

Reply:
Thanks. Will remove the unnecessary block.